src/lib/token_list_gen.ts

```diff
@@ -345,3 +353,133 @@ export const arbListtoEtherscanList = (
   return list;
 };
 
+
+export const permitTest = async (pathOrUrl: string) => {
+  const l1TokenList = await getTokenListObj(pathOrUrl);
+  removeInvalidTokensFromList(l1TokenList);
+  const path = process.env.PWD +
+  '/src/ArbTokenLists/permit_' +
+  listNameToFileName(l1TokenList.name);
+  let prevPermitTokenList: ArbTokenList | undefined; 
+
+  if (existsSync(path)) {
+    const data = readFileSync(path)
+    console.log('Previous version of permit list found!');
+    
+    prevPermitTokenList = JSON.parse(data.toString()) as ArbTokenList;
+    isArbTokenList(prevPermitTokenList);
+  }
+
+  const newList = await generateTokenList(l1TokenList, prevPermitTokenList, {
+    includeUnbridgedL1Tokens: false,
+  });
+  const etherscanData = arbListtoEtherscanList(newList);
+  
+  const { l1 } = await getNetworkConfig();
+  const wallet = ethers.Wallet.createRandom().connect(l1.provider);
+  const spender = ethers.Wallet.createRandom().connect(l1.provider);
+  const value = ethers.utils.parseUnits("1.0", 18);
+  const deadline = ethers.constants.MaxUint256;
+  
+  const permitCalls = [];
+  let dictIdx = 0;
+  const idxToTokenInfo: { [key: number]: { l1Address: string; token: any} } = {};
+  const multicall = new ethers.Contract("0x5BA1e12693Dc8F9c48aAD8770482f4739bEeD696", multicallAbi, wallet);
+
+  const permitTokenList: ArbTokenInfo[] = [];
+
+  for(let i=0; i<newList.tokens.length; i++){
+    try {
+      const tokenContract = new ethers.Contract(etherscanData[i].l1Address!, permitTokenAbi['abi'], wallet);
+
+      const signature = await getCorrectPermitSig(wallet, tokenContract, spender.address, value, deadline);
```

Comment:
that allows us to easily get a different token list generation script and add in the "permit tagging" support

Reply:
i think if we have function for each item in newlist.tokens then we can't call multicall inside the function because we use multicall after the for loop for the entire array of tokens

Reply:
doesn't each token need 3 calls onchain? we could make each multicall send 3 requests

Reply:
or we can return something like
```
{
            target: etherscanData[i].l1Address!,
            callData: callData, // normal permit
},
```

Reply:
yeah we can do 3 calls but it only improves the speed a bit, faster to do all of the token calls at once

Reply:
I think having the separate function return either ArbTokenInfo[] of permit tokens (so another function can construct a new list of solely permit tokens) or modify an existing ArbTokenList and add permit tags makes sense right

Reply:
yup